src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -419,6 +480,50 @@ static term start_ap(term ap_config, GlobalContext *global)
         }
     }
 
+    char *ap_hostname = driver_data->ap_hostname;
+    if (term_is_invalid_term(hostname_term)) {
+        ap_hostname = malloc(DEFAULT_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(ap_hostname)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            term error = get_default_device_name(ap_hostname, global);
+            if (error != OK_ATOM) {
+                free(ssid);
+                free(psk);
+                free(ap_hostname);
+                return error;
+            }
+        }
+    } else {
+        int ok = 0;
+        char *buf = malloc(MAX_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            buf = interop_term_to_string(hostname_term, &ok);
+        }
+        if (!ok || IS_NULL_PTR(buf)) {
```

Comment:
if ok is false, we still need to free buf.

Reply:
`(!ok || IS_NULL_PTR(buf))` might also be normalized to `(UNLIKELY(!ok || buf != NULL))`.

Reply:
I fixed both comments.